src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableHashSet_1.cs

```diff
@@ -754,8 +754,13 @@ private static bool Overlaps(IEnumerable<T> other, MutationInput origin)
         private static bool SetEquals(IEnumerable<T> other, MutationInput origin)
         {
             Requires.NotNull(other, nameof(other));
+            ISet<T>? otherSet = other switch
```

Comment:
```suggestion
            ISet<T> otherSet = other switch
```
Be explicit that `otherSet` is not null.